conda/environments/cudf_dev.yml

```diff
@@ -8,7 +8,6 @@ channels:
 dependencies:
 - cmake>=3.12
 - python>=3.6,<3.8
-- pytest
```

Comment:
why are we dropping pytest?

When making changes to cudf/python locally, I rely on this dev container being able to run pytest and confirm nothing broke (and that new tests pass)

Reply:
It was duplicated, it's defined again below.